### PR TITLE
[#770] Switch back from Mnesia to simple ETS

### LINKED ISSUE #770: Discussion : should we trash the db on startup?
At the moment, the startup process hangs while we wait for mnesia tables to be ready. For any decently sized code base, this wait can be quite long, in the order of minutes.

But we are now able to do indexing while the application is running, and index on-demand as a file is opened.  We also provide very clear feedback on the state of indexing vie `$/progress` messages.

So perhaps we will be better off just deleting the db on startup, so the IDE becomes responsive sooner, and the user can wait for indexing for full search capabilities.

Other possible alternatives are

a) use some other db instead of mnesia
b) use mnesia for core (small) information, and store the rest in files in the filesystem (perhaps using LSIF?).

`b)` is probably a very bad option, as it is basically rolling our own db.

### PR DESCRIPTION
**Pull request by robertoaloi**
### Description

For now, let's switch back to plain ETS. We can then follow up by adding persistence (via DETS) if necessary. In that case, we could have a dedicated option in the `erlang_ls.config` file to enable it. Before adding persistency, we should really check if it's worth, performance-wise.

All write operations to the ETS tables are serialized via a `gen_server`. Reads happen straight from the ETS table, which has `read_concurrency` enabled, even if it shouldn't matter much in Erlang LS.

Fixes #770 .

### CODE REVIEW

File: src/els_dt_document.erl [hunk 1]
```diff
@@ -72,12 +72,7 @@ name() -> ?MODULE.
 
 -spec opts() -> proplists:proplist().
 opts() ->
-  [ {attributes        , record_info(fields, els_dt_document)}
-  , {disc_copies       , [node()]}
-  , {index             , []}
-  , {type              , set}
-  , {storage_properties, [{ets, []}]}
-  ].
+  [].
```
Comment (alanz) [inline src/els_dt_document.erl:75]: Is `set` the default?  Should we be explicit?

**Review by alanz — APPROVED** [review]
LGTM

Given we now use unique node-names, I suggest removing the SHA value, and using the base of the project root instead.

**Comment by robertoaloi** [comment]
> LGTM
> 
> Given we now use unique node-names, I suggest removing the SHA value, and using the base of the project root instead.

Agreed, and we can also sync the DAP node name, which is today based on the SHA of the CWD. Will do it separately, though.

File: src/els_dt_document.erl (hunk 1 above)
Comment (robertoaloi) [reply src/els_dt_document.erl:75]: Yes, `set` is the default type. We can be explicit, will do it as a separate PR (I want to change something else).

**Pull request merged**